### PR TITLE
QR code login

### LINKED ISSUE #101: Login through QR codes
JsonLines has done excellent work and reverse engineered QR code authentication. This is a great way to bypass ENTs. 
https://github.com/Androz2091/pronote-qrcode-api

If we implement this, we should release it in v3.0.0, since we will probably have to break our current Client creation system.

### PR DESCRIPTION
**Pull request by Bapt5**
* Implement login with QR code 🎉 (fix #101)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,9 +89,33 @@ if client.logged_in:
...
+if client.logged_in:
+  # get the all the periods (may return multiple types like trimesters and semesters but it doesn't really matter
+  # the api will get it anyway)
+  periods = client.periods
+
+  for period in periods:
+      for grade in period.grades:  # iterate over all the grades
+          print(f'{grade.grade}/{grade.out_of}')  # print out the grade in this style: 20/20
+else:
+    print('no login')
```
Comment (bain3) [inline README.md:113]: I don't think we need the rest...
```suggestion
```

**Review by bain3 — CHANGES_REQUESTED** [review]
There's a mypy error but I think it is just the worker not having all deps. Great work! Good non-breaking implementation.

**Comment by Bapt5** [comment]
@bain3 I've a question about the roadmap What you mean by `Consistent ENT function naming` and `Split up dataClasses.py` ?

**Comment by bain3** [comment]
Afaik ent aren't all called the same. I'd change their name to `ent_something`.

`dataClasses.py` is getting long. Splitting it up is for the best. Making a file for each model for example.

**Comment by Bapt5** [comment]
> Afaik ent ne sont pas tous appelés de la même manière. Je changerais leur nom en `ent_something`.
> 
> `dataClasses.py`devient long. Le diviser est pour le mieux. Faire un dossier pour chaque modèle par exemple.

Okay I understand I will do that.

**Comment by bain3** [comment]
Oh noo. I think merging the qr login before 3.0.0 is ok. For 3.0.0 I'd like to clean up a bit in general. Can we make this pr only the qr?

**Comment by Bapt5** [comment]
Oh excuse me. II was thinking of doing all the possible updates I can do.

**Comment by Bapt5** [comment]
I don't know how I can remove my last commit from PR

**Comment by bain3** [comment]
Yeah I guess

**Comment by Bapt5** [comment]
I create a new branch on my repo to make all change for 3.0.0. It will be simpler

**Review by bain3 — APPROVED** [review]
LGTM

**Pull request merged**